src/lib.rs

```diff
@@ -360,131 +362,143 @@ impl RenderPass {
-            return;
-        }
-        // We need to convert the texture into rgba_srgb format.
-        let mut pixels: Vec<u8> = Vec::with_capacity(font_image.pixels.len() * 4);
-        for srgba in font_image.srgba_pixels(1.0) {
-            pixels.push(srgba.r());
-            pixels.push(srgba.g());
-            pixels.push(srgba.b());
-            pixels.push(srgba.a());
-        }
-        let font_image = egui::FontImage {
-            version: font_image.version,
-            width: font_image.width,
-            height: font_image.height,
-            pixels,
-        };
-        let bind_group = self.egui_texture_to_wgpu(device, queue, &font_image, "egui");
-
-        self.texture_version = Some(font_image.version);
-        self.texture_bind_group = Some(bind_group);
-    }
-
-    /// Updates the user textures that the app allocated. Should be called before `execute()`.
-    pub fn update_user_textures(&mut self, device: &wgpu::Device, queue: &wgpu::Queue) {
-        let pending_user_textures = std::mem::take(&mut self.pending_user_textures);
-        for (id, texture) in pending_user_textures {
-            let bind_group = self.egui_texture_to_wgpu(
-                device,
-                queue,
-                &texture,
-                format!("user_texture{}", id).as_str(),
-            );
-            self.user_textures.insert(id, bind_group);
-        }
-    }
-
-    /// Assumes font_image contains srgb data.
-    fn egui_texture_to_wgpu(
-        &self,
-        device: &wgpu::Device,
-        queue: &wgpu::Queue,
-        font_image: &egui::FontImage,
-        label: &str,
-    ) -> wgpu::BindGroup {
-        let size = wgpu::Extent3d {
-            width: font_image.width as u32,
-            height: font_image.height as u32,
-            depth_or_array_layers: 1,
-        };
+        textures: &egui::TexturesDelta,
+    ) -> Result<(), BackendError> {
+        for (texture_id, image_delta) in textures.set.iter() {
+            let image_size = image_delta.image.size();
+
+            let origin = match image_delta.pos {
+                Some([x, y]) => wgpu::Origin3d {
+                    x: x as u32,
+                    y: y as u32,
+                    z: 0,
+                },
+                None => wgpu::Origin3d::ZERO,
+            };
+
+            let alpha_srgb_pixels: Option<Vec<_>> = match &image_delta.image {
+                egui::ImageData::Color(_) => None,
+                egui::ImageData::Alpha(a) => Some(a.srgba_pixels(1.0).collect()),
+            };
+
+            let image_data: &[u8] = match &image_delta.image {
+                egui::ImageData::Color(c) => bytemuck::cast_slice(c.pixels.as_slice()),
+                egui::ImageData::Alpha(_) => {
+                    // The unwrap here should never fail as alpha_srgb_pixels will have been set to
+                    // `Some` above.
+                    bytemuck::cast_slice(alpha_srgb_pixels.as_ref().unwrap().as_slice())
```

Comment:
Even if it's impossible in the current state of the code, don't throw a generic panic and at least specify the panic reason  with "expect()".